ctapipe/io/tableloader.py

```diff
@@ -667,5 +674,76 @@ def read_telescope_events_by_type_chunked(self, chunk_size, **kwargs):
             self.read_telescope_events_by_type,
             n_total=len(self),
             chunk_size=chunk_size,
-            **kwargs,
+            args=args,
+            kwargs=kwargs,
+        )
+
+    def read_telescope_events_by_id(
+        self, telescopes=None, start=None, stop=None
+    ) -> Dict[int, Table]:
+        """Read telescope-based event information.
+
+        Parameters
+        ----------
+        telescopes: List[Union[int, str, TelescopeDescription]]
+            Any list containing a combination of telescope IDs or telescope_descriptions.
+
+        Returns
+        -------
+        tables: dict(astropy.io.Table)
+            Dictionary of tables organized by telescope ids
+            Table with primary index columns "obs_id", "event_id" and "tel_id".
+        """
+
+        if telescopes is None:
+            tel_ids = tuple(self.subarray.tel.keys())
+        else:
+            tel_ids = self.subarray.get_tel_ids(telescopes)
+
+        subarray_events = self.read_subarray_events(
+            start=start, stop=stop, keep_order=False
+        )
+        self._add_index_if_needed(subarray_events)
+
+        tel_start, tel_stop = self._get_tel_start_stop(tel_ids, start, stop)
+        tel_start = tel_start if tel_start is not None else [None] * len(tel_ids)
+        tel_stop = tel_stop if tel_stop is not None else [None] * len(tel_ids)
+
+        by_id = {}
+        for tel_id, start, stop in zip(tel_ids, tel_start, tel_stop):
+            # no events for this telescope in range start/stop
+            if start is not None and stop is not None and (stop - start) == 0:
+                continue
+
+            by_id[tel_id] = self._read_telescope_events_for_id(
+                tel_id, start=start, stop=stop
+            )
+
+        for tel_id in by_id.keys():
+            by_id[tel_id] = self._join_subarray_info(
+                by_id[tel_id], subarray_events=subarray_events
+            )
+            self._sort_to_original_order(by_id[tel_id], include_tel_id=True)
+
+        return by_id
+
+    def read_telescope_events_by_id_chunked(self, chunk_size, *args, **kwargs):
```

Comment:
Very similar to `read_telescope_events_by_type_chunked`. Perhaps one function would be sufficient
```python
     def read_telescope_events_by_chunked(self, chunk_size, events_by, *args, **kwargs):       
         """                                                                                   
         Iterate over chunks of telescope events and return a dict of one table per telescope id.    
                                                                                               
         Parameters                                                                            
         ----------                                                                            
         chunk_size: int                                                                       
             Number of subarray events to load per chunk.                                      
             The telescope tables might be larger or smaller than chunk_size                   
             depending on the selected telescopes.                                             
                                                                                               
         *args, **kwargs are passed to `read_telescope_events_by_id`                           
         """                                                                                   
         return ChunkIterator(                                                                 
             self.__getattribute__(f'read_telescope_events_by_{event_by}),                    
             n_total=len(self),                                                                
             chunk_size=chunk_size,                                                            
             args=args,                                                                        
             kwargs=kwargs,                                                                    
         )              
```

Reply:
It's more about having the proper docstring and being discoverable in the documentation, I'd say. The code duplicated is very little.

Reply:
ok